src/commands/help.ts

```diff
@@ -1,9 +1,43 @@
+import { EmbedBuilder } from 'discord.js';
+
 export const help: GlobalCommand = {
 	name: 'help',
 	description: 'Prints the list of commands',
 	requiresGuild: false,
-	async execute({ reply }) {
-		// TODO: finish this command
-		await reply('Hello, world!');
+	async execute({ guild, reply }) {
+		// Dynamic import here b/c ./index depends on this file
+		const { allCommands } = await import('./index');
+
+		const embed = new EmbedBuilder() //
+			.setTitle('All commands');
+
+		function embedCommand(command: Command): void {
+			embed.addFields({
+				name: `\`/${command.name}\``, // i.e. `/help`
+				value: command.description,
+			});
+		}
+
+		for (const command of allCommands.values()) {
+			if (guild) {
+				// We're in a guild. We should check the user's permissions,
+				// and skip this command if they aren't allowed to use it.
+				// TODO: Grab the guild's configured permissions for this channel
+				embedCommand(command);
+				continue;
+			}
```

Comment:
Is this TODO being worked on at the moment?

Reply:
Or is the building of perms checking something that we should add to the docket and circle back to?

Reply:
Might be complicated. I poked at the problem a bit, and I realized I need to learn how this new guild-centric permission system works. I'm used to the bot maintaining a database of its own command permissions, but slash commands don't do that anymore, so I need to re-learn how the APIs relate to the UI.

If someone can pick that up before I get around to it, be my guest. Meantime, since Discord enforces command permissions anyway, I'm fine pushing this one down the road.

Reply:
Alright. I think we can revisit this when we get a little more experience with the API then. I'll add is as a ticket for future building.